specs/protocol/safer-safes.md

```diff
@@ -0,0 +1,507 @@
+- MUST revert if there isn't a challenge for the calling `safe`.
+- MUST reset `challenge_start_time` to 0.
+- MUST emit the `ChallengeCancelled` event.
+
+### `changeOwnershipToFallback`
+
+With a successful challenge, removes all current owners from an enabled `safe`, appoints `fallback` as its sole owner,
+and sets its quorum to 1.
+
+- MUST be executable by anyone.
+- MUST revert if the `safe` hasn't enabled the contract as a module.
+- MUST revert if the `safe` hasn't configured the contract as a module.
+- MUST revert if there isn't a successful challenge for the given `safe`.
+- MUST reset `challenge_start_time` to 0 to enable the fallback to start a new challenge.
+- MUST emit the `ChallengeExecuted` event.
+
+### `cancellationThreshold`
+
+Returns the `cancellation_threshold` for a given safe.
+
+- MUST NOT revert
+- MUST return 0 if the contract is not enabled as a guard for the safe.
+
+### `scheduleTransaction`
+
+Called by anyone using signatures from Safe owners, registers a transaction in the TimelockGuard for execution after
+the `timelock_delay`.
+
+- MUST revert if the contract is not enabled as a guard for the safe.
+- MUST take the same parameters as `execTransaction`.
+- MUST revert if an identical transaction has already been scheduled.
+- MUST revert if an identical transaction was cancelled.
+- MUST set `execution_time(safe, tx)` to `block.timestamp + timelock_delay(safe)`.
+- MUST emit a `TransactionScheduled` event, with at least `safe` and relevant transaction data.
+
+To allow for identical transactions to be scheduled more than once, but requiring different signatures for each one, a
+`salt` parameter can be included in `data` with the sole purpose of differentiating otherwise identical transactions.
+
+### `checkTransaction`
+
+Called by anyone, and also by the Safe in `execTransaction`, verifies if a given transaction was scheduled and the
+delay period has passed.
+
+- MUST revert if the contract is not enabled as a guard for the safe.
+- MUST take the exact parameters from the `ITransactionGuard.checkTransaction` interface.
+- MUST revert if `execution_time(safe, tx) < block.timestamp`.
+- MUST revert if the scheduled transaction was cancelled.
+
+### `checkPendingTransactions`
+
+Called by anyone, returns the list of all scheduled but not cancelled transactions for a given safe.
+
+- MUST NOT revert
+
+*Note:* If we want to exclude executed transactions from this list, we would need to implement the
+`checkAfterExecution` hook and store the executed state for transactions.
+
+### `rejectTransaction`
+
+Called by a Safe owner, signal the rejection of a scheduled transaction.
+
+- MUST revert if the contract is not enabled as a guard for the safe.
+- MUST revert if not called by an owner of the safe scheduled to execute the transaction, or in one of its child safes.
+- MUST be able to reference a transaction scheduled in a different safe. The transaction might not exist.
```

Comment:
I don't understand what this requirement means?

Reply:
If you have a nested setup, transactions will be scheduled in the parent safe, but rejected in the child safes.

I think I wrote this without considering that the guard is a singleton, and was worried that the timelock guard in the child safe wouldn't have access to the storage for the parent safe timelock guard.

Given that the timelock guard is a singleton, and all transactions have an unique identifier regardless of the safe scheduled for execution, this doesn't make much sense.